src/metabase/lib/convert.cljc

```diff
@@ -40,6 +42,10 @@
                                               (map (juxt :type :in))
                                               first)]
       (let [new-stage (clean-location almost-stage error-type error-location)]
+        (log/warnf "Clean: Removing bad clause in %s due to error %s:\n%s"
+                   (u/colorize :yellow (pr-str error-location))
+                   (u/colorize :yellow (pr-str error-type))
+                   (u/colorize :red (u/pprint-to-str (first (data/diff almost-stage new-stage)))))
```

Comment:
I added a log warning if the clean logic removes something, it makes it easier to debug stuff.